[[ Bug 17958 ]] Ignore lcidl 'tail' attribute on desktop

runrevmark opened the pull request:
Tail handlers specified in LCIDL should only use 'run on main thread'
on mobile platforms.

runrevmark committed:
[[ Bug 17958 ]] Ignore lcidl 'tail' attribute on desktop

Tail handlers specified in LCIDL should only use 'run on main thread'
on mobile platforms.

peter-b commented:
@livecode-vulcan review ok c36474b

livecode-vulcan commented:
💙 review by @peter-b ok c36474ba6c4f7497ffc02c5d4447a6cf95516712

livecode-vulcan commented:
😎 test success c36474ba6c4f7497ffc02c5d4447a6cf95516712
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2754)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2763)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2749)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2765)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2771)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2775)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2793)

montegoulding commented:
@runrevmark I think LCObjectPost will still be broken on server with this patch.

runrevmark committed:
[[ Bug 17958 ]] Make sure run on main thread works on server

This patch compiles the desktop implementation of run on main thread
in server. This should mean things like LCObjectPost will work, as
long as the script runs a simple message dispatch loop:

    repeat forever
        wait for messages
    end repeat

montegoulding commented:
@livecode-vulcan review ok 772b0b1

livecode-vulcan commented:
💙 review by @montegoulding ok 772b0b188372f0045d2704901fa1fc6bc8913709

livecode-vulcan commented:
😎 test success 772b0b188372f0045d2704901fa1fc6bc8913709
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2760)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2769)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2755)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2771)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2777)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2781)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2799)
